tests/emmc.c

```diff
@@ -42,6 +42,90 @@ emmc_simple(void)
     g_assert_cmpint(g_rmdir(path), ==, 0);
 }
 
+static void
+test_raw_overwrite_pass(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autoptr(PuConfig) config = NULL;
+    g_autoptr(PuEmmc) emmc = NULL;
+
+    config = pu_config_new_from_file("config/raw-non-overlap.yaml", &error);
+    g_assert_nonnull(config);
+
+    g_autofree gchar *path = NULL;
+    g_autoptr(GFile) mmcblk0 = NULL;
+    path = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+    mmcblk0 = create_tmp_file("mmcblk0", path, 100 * PED_MEBIBYTE_SIZE, &error);
+
+    emmc = pu_emmc_new(g_file_get_path(mmcblk0), config, "data", FALSE, &error);
+    g_assert_no_error(error);
+    g_assert_nonnull(emmc);
+}
+
+static void
+test_raw_overwrite_fail_partition_table(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autoptr(PuConfig) config = NULL;
+    g_autoptr(PuEmmc) emmc = NULL;
+
+    config = pu_config_new_from_file("config/raw-overlap-partition-table.yaml", &error);
+    g_assert_nonnull(config);
+
+    g_autofree gchar *path = NULL;
+    g_autoptr(GFile) mmcblk0 = NULL;
+    path = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+    mmcblk0 = create_tmp_file("mmcblk0", path, 100 * PED_MEBIBYTE_SIZE, &error);
+
+    emmc = pu_emmc_new(g_file_get_path(mmcblk0), config, "data", FALSE, &error);
+    g_assert_error(error, PU_ERROR, PU_ERROR_EMMC_PARSE);
+    g_assert_null(emmc);
+}
+
+static void
+test_raw_overwrite_fail_partition(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autoptr(PuConfig) config = NULL;
+    g_autoptr(PuEmmc) emmc = NULL;
+
+    config = pu_config_new_from_file("config/raw-overlap-partition.yaml", &error);
+    g_assert_nonnull(config);
+
+    g_autofree gchar *path = NULL;
+    g_autoptr(GFile) mmcblk0 = NULL;
+    path = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+    mmcblk0 = create_tmp_file("mmcblk0", path, 100 * PED_MEBIBYTE_SIZE, &error);
+
+    emmc = pu_emmc_new(g_file_get_path(mmcblk0), config, "data", FALSE, &error);
+    g_assert_error(error, PU_ERROR, PU_ERROR_FAILED);
+    g_assert_null(emmc);
+}
+
+static void
+test_raw_overwrite_fail_raw(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autoptr(PuConfig) config = NULL;
+    g_autoptr(PuEmmc) emmc = NULL;
+
+    config = pu_config_new_from_file("config/raw-overlap-raw.yaml", &error);
+    g_assert_nonnull(config);
+
+    g_autofree gchar *path = NULL;
+    g_autoptr(GFile) mmcblk0 = NULL;
+    path = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+    mmcblk0 = create_tmp_file("mmcblk0", path, 100 * PED_MEBIBYTE_SIZE, &error);
```

Comment:
The temporary directory is not removed after the test has been run. I think you should use a fixture for these tests. Maybe `EmptyDeviceFixture` could be used.

Reply:
The `EmptyDeviceFixture` can be used here but would require root to work, which is something I'd like to avoid. The `EmptyFileFixture` is another option but has currently no way to specify the file name which is required here.

Reply:
Maybe that's were the `user_data` variable might come in handy? You could pass it the filename.